static/js/hoc/withInfiniteScroll.js

```diff
@@ -0,0 +1,129 @@
+// @flow
+import React from "react"
+import qs from "query-string"
+import InfiniteScroll from "react-infinite-scroller"
+
+import { Loading } from "../components/Loading"
+import PostList from "../components/PostList"
+
+import { toggleUpvote } from "../util/api_actions"
+import { actions } from "../actions"
+
+import type { Location } from "react-router"
+import type { Dispatch } from "redux"
+import type { Post, PostListPagination } from "../flow/discussionTypes"
+
+type Props = {
+  canLoadMore: boolean,
+  isModerator: boolean,
+  loadPosts: (search: PostListPagination) => Promise<*>,
+  location: Location,
+  pagination: PostListPagination,
+  posts: ?Array<Post>,
+  reportPost: (p: Post) => void,
+  removePost: (p: Post) => void,
+  toggleUpvote: (post: Post) => Promise<*>,
+  dispatch: Dispatch<any>,
+  showChannelLinks: boolean,
+  showReportPost: boolean,
+  showRemovePost: boolean,
+  showTogglePinPost: boolean,
+  showPinUI: boolean
+}
+
+const withInfiniteScroll = (WrappedComponent: Class<React.Component<*, *>>) => {
```

Comment:
I would call this `withPostList` or `withPostDisplay` or something. The purpose of this HOC is to handle the logic of rendering a list of posts. The fact that the posts are navigable via infinite scroll is an implementation detail